pgo/workloads/cs-example/run

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash -eu
+
+trap 'pkill -P $$ || true' EXIT
+
+DIR=`dirname $(realpath "$0")`
+BIN=`realpath "$1"`
+TYPE="$2"
+OUT=`realpath "$3"`
+cd "$DIR"
+
+function wait_for_port {
+    timeout --foreground "$3" bash -c 'until printf "" 2>>/dev/null >>/dev/tcp/'"$1/$2"'; do sleep 0.1 || exit 1; done'
+}
+
+SERVER_ADDRS=(127.42.42.1 127.42.42.2 127.42.42.3)
+SERVER_PIDS=()
+
+for ADDR in "${SERVER_ADDRS[@]}"; do
+    cd "$OUT" # chdir to OUT so that bolt instrumentation leaves its output there
+    env LLVM_PROFILE_FILE="$OUT/$ADDR-%m.profraw" "$BIN" --options-file="$DIR/scylla.yaml" --workdir="$OUT"/workdir-"$ADDR" --ring-delay-ms=1000 --overprovisioned --developer-mode=yes --smp=2 --memory=1G --seed-provider-parameters=seeds="${SERVER_ADDRS[0]}" --listen-address="$ADDR" --rpc-address="$ADDR" --api-address="$ADDR" 2>"$OUT/$ADDR.log" ${BOLT_OPTION:-} &
+    cd -
+    if ! wait_for_port "$ADDR" 7000 60; then
+        echo "Server "$ADDR" did not set up the RPC socket in 60 seconds. Aborting" 1>&2
+        exit 1
+    fi
+    SERVER_PIDS+=($!)
+done
+for ADDR in "${SERVER_ADDRS[@]}"; do
+    if ! wait_for_port "$ADDR" 9042 60; then
+        echo "Server "$ADDR" did not set up the CQL socket in 60 seconds. Aborting" 1>&2
+        exit 1
+    fi
+done
+
+printf -v PIDLIST '%s,' "${SERVER_PIDS[@]}"
+
+DURATION=30s
+N_PARTITIONS=10000000
+../../../tools/java/tools/bin/cassandra-stress write n="$N_PARTITIONS" cl=quorum no-warmup -mode native cql3 -node 127.42.42.1
+../../../tools/java/tools/bin/cassandra-stress mixed 'ratio(write=1,read=1)' cl=quorum no-warmup duration="$DURATION" -pop dist="UNIFORM(1..$N_PARTITIONS)" -mode native cql3 -node 127.42.42.1 -rate threads=200
+
```

Comment:
> Are you sure that this load is really "representative"?

This load is an example/placeholder. It's not supposed to be used as the basis for PGO in this shape. The representative set of workloads is not worked out yet.

> In particular it's not obvious to me that you are going to trigger at least one memtable flush and a compaction event after 30 seconds.

Flushes and compactions should be taken care of by the population phase (which doesn't have a specified duration, it just writes 10000000 rows).

> Please, note that Scylla's performance naturally suffers when there are background processes running: things like compaction, repair, streaming, etc.
> Therefore you need to at least make sure that while optimizing the "best case" scenario you don't make the "worst case" scenario even worse.

Indeed. I meant to address this concern via the commit message of "pgo: introduce PGO scripts". Citing:

> The quality of PGO is correlated with the quality of the profile.
> As a rule, the closer the profiling workloads are to the production workload,
> the greater the gain from PGO.
> 
> Ideally all code that is hot in some production workload should be exercised
> in the profile. Untouched code may be treated by the optimizer as cold
> and e.g. get optimized for size (that's what the -pgso knob in LLVM does).
> If this happens to actually hot code, it could in theory cause a performance
> regression around that piece of code.
> 
> The ultimate profile can be merged (possibly with some weights) from multiple
> profiles from multiple runs. Currently pgo/run_all delegates the profile
> collection to workload subdirectories and merges the results without weights.
> 
> As of this commit, there is only one subdirectory - cs_example - included.
> It loads Scylla with the base read/write workload of cassandra-stress.

Yes, it's theoretically possible to cause regressions by using a profile that's not good enough. I think the best way to assess the potential impact of that is to evaluate the worst (most mismatched) training/test combination we can come up with.

(My expectation is that regressions due to badly-trained PGO are very unlikely, especially if we preventively disable -pgso, which AFAIK could be the main source of regressions in this situation).